StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Elements/LinkPaymentMethodFormElement.swift

```diff
@@ -233,16 +216,32 @@ final class LinkPaymentMethodFormElement: Element {
     private lazy var billingAddressSection: AddressSectionElement? = {
         guard configuration.billingDetailsCollectionConfiguration.address != .never else { return nil }
 
+        let collectPhone = configuration.billingDetailsCollectionConfiguration.phone == .always && isBillingDetailsUpdateFlow
```

Comment:
The phone number should only be collected in the billing details update flow, since it'll be passed to the confirmation call, not to the update call.